### PR TITLE
[16.0][MIG] stock_warehouse_flow: Migration to 16.0

### PR DESCRIPTION
**Pull request by RodrigoBM**
#461

### CODE REVIEW

**Comment by RodrigoBM** [comment]
test do not pass, this https://github.com/OCA/wms/pull/707 is necessary before

**Review by xAdrianCif — APPROVED** [review]
Technical review

**Comment by RodrigoBM** [comment]
Add this fix https://github.com/OCA/wms/pull/705

**Review by jsanchezdeavila — APPROVED** [review]
Functional validation

LGTM 

https://www.awesomescreenshot.com/video/19771910?key=4864abd489cf337b68d79215091357ab

File: stock_warehouse_flow/models/stock_warehouse_flow.py [hunk 1]
```diff
@@ -342,14 +346,16 @@ def _generate_delivery_route(self):
                         }
                     )
                 wh_vals["out_type_id"] = self.to_picking_type_id.id
-                self.warehouse_id.write(wh_vals)
+                self.warehouse_id.with_context(do_not_check_quant=True).write(wh_vals)
```
Comment (sebalix) [inline stock_warehouse_flow/models/stock_warehouse_flow.py:348]: I know this change is coming from 14.0, but having a comment explaining the reason of this context key is very welcome here (in another commit to be easily backported if needed).
Note: for new commits better to avoid writing the version number in the message, it makes sense only for migration commit ("Migration to X.Y") but others commits could be backported/forward ported across versions :) Thanks!

File: stock_warehouse_flow/models/stock_route.py [hunk 2]
```diff
@@ -4,8 +4,8 @@
 from odoo import api, fields, models
 
 
-class StockLocationRoute(models.Model):
-    _inherit = "stock.location.route"
+class StockRoute(models.Model):
+    _inherit = "stock.route"
 
     flow_id = fields.Many2one(
```
Comment (sebalix) [inline stock_warehouse_flow/models/stock_route.py]: Since the data model has been renamed by Odoo, do you know if we are losing the `flow_id` column in the process or it's all good? (with OpenUpgrade it should be OK, but with Odoo S.A. I have no feedback yet).
This field is set at line 363 in `_generate_delivery_route` method, and used in `_compute_impacted_route_ids` at line 187.

File: stock_warehouse_flow/views/stock_warehouse_flow.xml [hunk 3]
```diff
@@ -16,30 +16,35 @@
...
+                <field
+                                    name="warehouse_id"
+                                    attrs="{'readonly': [('delivery_route_id', '!=', False)]}"
+                                />
+              </group>
+              <group>
+                <field name="sequence" />
+                <field name="active" invisible="1" />
+                <field name="company_id" invisible="1" />
+              </group>
```
Comment (sebalix) [inline stock_warehouse_flow/views/stock_warehouse_flow.xml:36]: Not blocking. Are these view changes required for the migration? (Odoo changes the way it is rendered and without it it's "ugly"?).
If not this deserves to be in its own commit to be backported.

File: stock_warehouse_flow/views/stock_warehouse_flow.xml [hunk 4]
```diff
@@ -16,30 +16,35 @@
...
                             >
-                <span style="width: 4em;" />
-                =>
-                <field name="from_location_src_id" nolabel="1" />
-                to
-                <field name="from_location_dest_id" nolabel="1" />
+                <span class="text-muted"> => </span>
+                <field name="from_location_src_id" nolabel="1" class="oe_inline" />
+                <span class="text-muted"> to </span>
+                <field name="from_location_dest_id" nolabel="1" class="oe_inline" />
```
Comment (sebalix) [inline stock_warehouse_flow/views/stock_warehouse_flow.xml]: Same question.

File: stock_warehouse_flow/views/stock_warehouse_flow.xml (hunk 3 above)
Comment (RodrigoBM) [reply stock_warehouse_flow/views/stock_warehouse_flow.xml:36]: Yes, these changes are necessary for migration.

View in v14:
Selección_492

View without changes in V16:
Selección_491

View with changes in V16:
Selección_490

File: stock_warehouse_flow/views/stock_warehouse_flow.xml (hunk 4 above)
Comment (RodrigoBM) [reply stock_warehouse_flow/views/stock_warehouse_flow.xml]: same answer

File: stock_warehouse_flow/views/stock_warehouse_flow.xml (hunk 3 above)
Comment (sebalix) [reply stock_warehouse_flow/views/stock_warehouse_flow.xml:36]: Thank you for the feedback!

File: stock_warehouse_flow/models/stock_route.py (hunk 2 above)
Comment (RodrigoBM) [reply stock_warehouse_flow/models/stock_route.py]: I don't know, we haven't tested any migration yet as our customers are new to v16. I understand that there should be no problems.

File: stock_warehouse_flow/models/stock_warehouse_flow.py (hunk 1 above)
Comment (RodrigoBM) [reply stock_warehouse_flow/models/stock_warehouse_flow.py:348]: I don't know what you need here. I just add an approved fix in v14  https://github.com/OCA/wms/pull/705.

About the note: I think it is only necessary if the module is already migrated.

File: stock_warehouse_flow/models/stock_route.py (hunk 2 above)
Comment (sebalix) [reply stock_warehouse_flow/models/stock_route.py]: Okay, that was to get a feedback, thank you!

File: stock_warehouse_flow/models/stock_warehouse_flow.py (hunk 1 above)
Comment (sebalix) [reply stock_warehouse_flow/models/stock_warehouse_flow.py:348]: No problem!

**Review by sebalix — APPROVED** [review]

**Comment by sebalix** [comment]
Thank you @RodrigoBM

**Comment by RodrigoBM** [comment]
> Thank you @RodrigoBM

You are welcome.
